massgov/mayflower
# DP-9272 [a11y] Add aria-described by to directions in contact group address

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)



## Description

Set up an id for `item.label` to associate with "aria-describedby" in decorative link for its "directions" link to add link context in Contact.

When `item.label` value is available, a unique ID is added to `.ma__contact-group__label`.  Its ID value is passed to its associated decorative link for "directions" as `aria-describedby` value.  Thus, they establish an association between the location name and its "directions" link for AT.  AT users would clearly know what the "directions" link is for.

Decorative link wouldn't have `aria-describedby` when its value is not available.  

No visual change.

This change doesn't involve any Drupal change.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-9272)

## Steps to Test


Test in Drupal.  The org page in Mayflower doesn't have `item.label` which is necessary data for the component for the test.

1. Build the site in Drupal with this branch: `patternlab/DP-9272_direction-link`
2. Go to the sample test page. `/orgs/office-of-the-governor`.
3. Check the source for a contact item in the Contact Us.  Find: 
     - `.ma__contact-group__label` has id.  Its value starts with "linkContext-" followed by unique 8 digits.  
     - Its associated "directions" link has `aria-describedby` attribute with the same ID value.  
See the screenshots below.  Decorative link wouldn't have `aria-describedby` when its value is not available.  
Find each directions link has their associated location name for the link context:

         - Governor Charlie Baker's Office of Constituent Services
         - Western Massachusetts Office
         - Washington, DC Office

## Screenshots
Before
<img width="478" alt="before" src="https://user-images.githubusercontent.com/9633303/45162625-66b66e80-b1bc-11e8-802c-d9d25fcd5f60.png">

After
<img width="666" alt="after" src="https://user-images.githubusercontent.com/9633303/45164966-bc414a00-b1c1-11e8-9c33-66a6f2a12366.png">

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
